src/test/java/seedu/guest/logic/commands/MarkRoomUncleanCommandTest.java

```diff
@@ -0,0 +1,72 @@
+package seedu.guest.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.guest.logic.commands.CommandTestUtil.VALID_IS_ROOM_CLEAN_BOB;
+import static seedu.guest.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.guest.testutil.TypicalGuests.getTypicalGuestBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.guest.model.Model;
+import seedu.guest.model.ModelManager;
+import seedu.guest.model.UserPrefs;
+import seedu.guest.model.guest.Guest;
+import seedu.guest.testutil.GuestBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for MarkRoomUncleanCommand.
+ */
+public class MarkRoomUncleanCommandTest {
+
+    private Model model = new ModelManager(getTypicalGuestBook(), new UserPrefs());
+
+    @Test
+    public void execute_markRoomUncleanCommand_success() {
+        Guest guestInFilteredList0 = model.getFilteredGuestList().get(0);
+        Guest guestInFilteredList1 = model.getFilteredGuestList().get(1);
+        Guest guestInFilteredList2 = model.getFilteredGuestList().get(2);
+        Guest guestInFilteredList3 = model.getFilteredGuestList().get(3);
+        Guest guestInFilteredList4 = model.getFilteredGuestList().get(4);
+        Guest guestInFilteredList5 = model.getFilteredGuestList().get(5);
+        Guest guestInFilteredList6 = model.getFilteredGuestList().get(6);
+
+        Guest editedGuest0 = new GuestBuilder(guestInFilteredList0).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest1 = new GuestBuilder(guestInFilteredList1).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest2 = new GuestBuilder(guestInFilteredList2).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest3 = new GuestBuilder(guestInFilteredList3).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest4 = new GuestBuilder(guestInFilteredList4).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
+        Guest editedGuest5 = new GuestBuilder(guestInFilteredList5).withIsRoomClean(VALID_IS_ROOM_CLEAN_BOB).build();
```

Comment:
Should we have some cleaned and some uncleaned to ensure the command works for both?

Reply:
This part is to construct to the expectedModel. The model to be tested has both cleaned and uncleaned rooms.

Reply:
I see, thanks!